engine/src/graphics/device.cpp

```diff
@@ -16,7 +16,11 @@
 
 namespace gen
 {
+#ifdef GEN_PLATFORM_APPLE
+	static const std::vector<const char *> deviceExtensions = {VK_KHR_SWAPCHAIN_EXTENSION_NAME, VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME};
+#else
 	static const std::vector<const char *> deviceExtensions = {VK_KHR_SWAPCHAIN_EXTENSION_NAME};
+#endif
```

Comment:
It's safer to keep as much code as possible outside ifdefs, minimize chances of something unexpected changing later. 
```cpp
static const std::vector<const char *> deviceExtensions = {
    VK_KHR_SWAPCHAIN_EXTENSION_NAME, 
#ifdef GEN_PLATFORM_APPLE
    VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME,
#endif
};
```